src/bin/cobalt/new.rs

```diff
@@ -406,7 +408,31 @@ pub fn rename_document(
     Ok(())
 }
 
-pub fn publish_document(file: &path::Path) -> Result<()> {
+fn prepend_date_to_filename(
+    config: &cobalt_model::Config,
+    file: &path::Path,
+    date: &cobalt_model::DateTime,
+) -> Result<()> {
+    // avoid prepend to existing date prefix
+
+    let file_stem = cobalt_model::file_stem(file);
+    let (_, file_stem) = cobalt_model::parse_file_stem(file_stem);
+    let file_name = format!(
+        "{}{}.{}",
+        (**date).format("%Y-%m-%d-"),
+        file_stem,
+        file.extension()
+            .map(|os| os
+                .to_str()
+                .unwrap_or_else(|| &config.posts.template_extensions[0]))
+            .unwrap_or_else(|| &config.posts.template_extensions[0])
+    );
+    trace!("Prefix with date, new filename: {}", file_name);
```

Comment:
If we keep this trace, we should note the config setting that caused this so users can be helped along to disabling it if desired.